python/GafferOSLTest/OSLObjectTest.py

```diff
@@ -236,5 +236,137 @@ def testReferencePromotedPlug( self ) :
 
 		s["r"]["p"].setInput( s["s"]["out"] )
 
+	def testCanShadeVertexInterpolatedPrimitiveVariablesAsUniform( self ) :
+
+		s = Gaffer.ScriptNode()
+
+		c = GafferScene.Cube()
+		s.addChild( c )
+
+		o = GafferOSL.OSLObject()
+		s.addChild( o )
+
+		f = GafferScene.PathFilter( "PathFilter" )
+		s.addChild( f )
+		f["paths"].setValue( IECore.StringVectorData( [ '/cube' ] ) )
+		o["filter"].setInput( f["out"] )
+
+		# ensure the source position primitive variable interpolation is ver
+		self.assertEqual(c["out"].object("/cube")['P'].interpolation, IECore.PrimitiveVariable.Interpolation.Vertex )
+
+		o['in'].setInput( c["out"] )
+		o['interpolation'].setValue( IECore.PrimitiveVariable.Interpolation.Uniform )
+
+		inPoint = GafferOSL.OSLShader( "InPoint" )
+		s.addChild( inPoint )
+		inPoint.loadShader( "ObjectProcessing/InPoint", keepExistingValues = True )
+
+		vectorAdd = GafferOSL.OSLShader( "VectorAdd" )
+		s.addChild( vectorAdd )
+		vectorAdd.loadShader( "Maths/VectorAdd", keepExistingValues = True )
+		vectorAdd["parameters"]["b"].setValue( IECore.V3f( 1, 2, 3 ) )
+
+		vectorAdd["parameters"]["a"].setInput( inPoint["out"]["value"] )
+
+		outPoint = GafferOSL.OSLShader( "OutPoint" )
+		s.addChild( outPoint )
+		outPoint.loadShader( "ObjectProcessing/OutPoint", keepExistingValues=True )
+		outPoint['parameters']['name'].setValue("P_copy")
+
+		outPoint["parameters"]["value"].setInput( vectorAdd["out"]["out"] )
+
+		outObject = GafferOSL.OSLShader( "OutObject" )
+		s.addChild( outObject )
+		outObject.loadShader( "ObjectProcessing/OutObject", keepExistingValues=True )
+		outObject["parameters"]["in0"].setInput( outPoint["out"]["primitiveVariable"] )
+		outObject.loadShader( "ObjectProcessing/OutObject", keepExistingValues=True )
+
+		o["shader"].setInput( outObject["out"] )
+
+		cubeObject = s['OSLObject']['out'].object( "/cube" )
+
+		self.assertTrue( "P_copy" in cubeObject.keys() )
+		self.assertEqual( cubeObject["P_copy"].interpolation, IECore.PrimitiveVariable.Interpolation.Uniform)
+
+		self.assertEqual( cubeObject["P_copy"].data[0], IECore.V3f(  0.0,  0.0, -0.5 ) + IECore.V3f( 1, 2, 3 ))
+		self.assertEqual( cubeObject["P_copy"].data[1], IECore.V3f(  0.5,  0.0,  0.0 ) + IECore.V3f( 1, 2, 3 ))
+		self.assertEqual( cubeObject["P_copy"].data[2], IECore.V3f(  0.0,  0.0,  0.5 ) + IECore.V3f( 1, 2, 3 ))
+		self.assertEqual( cubeObject["P_copy"].data[3], IECore.V3f( -0.5,  0.0,  0.0 ) + IECore.V3f( 1, 2, 3 ))
+		self.assertEqual( cubeObject["P_copy"].data[4], IECore.V3f(  0.0,  0.5,  0.0 ) + IECore.V3f( 1, 2, 3 ))
+		self.assertEqual( cubeObject["P_copy"].data[5], IECore.V3f(  0.0, -0.5,  0.0 ) + IECore.V3f( 1, 2, 3 ))
+
+def testCanShadeFaceVaryingInterpolatedPrimitiveVariablesAsVertex( self ) :
```

Comment:
> I'm unclear if I need to update affects & hash to consider the resampledInPlug?

The best rule of thumb is to work backwards from the `compute()`, since that's the thing you're actually interested in implementing, and the rest is just an annoyance needed to keep caching and dirty propagation working. You only need to implement `compute()` for output plugs, and only then if they don't have an input connection. Since `resampledInPlug()` is an input plug _and_ has an input connection, you definitely don't need to implement compute for it and therefore don't need to compute a `hash()` for it, or add it to the affected outputs in `affects()`.

What I was trying to get it is that because you now access a value from `resampledInPlug()` in `computeProcessedObject()`, you also need to include it in the hash in `hashProcessedObject()` and ensure that it causes `out.object` to be dirtied in `affects()`. You're probably getting away with it for now because `interpolationPlug()` is approximating the same effect, but as we make the logic for which variables are resampled more complex, getting this right will become more important.